azure-pipelines/end-to-end-tests-dir/versions.ps1

```diff
@@ -2,18 +2,34 @@
 
 $versionFilesPath = "$PSScriptRoot/../e2e_ports/version-files"
 
+# create custom registry for testing
+$RegistryRootComplete = Join-Path $TestingRoot 'registry_for_versions_complete'
+& $GitExe init $RegistryRootComplete
+Throw-IfFailed
+$RegistryRootIncomplete = Join-Path $TestingRoot 'registry_for_versions_incomplete'
+& $GitExe init $RegistryRootIncomplete
+Throw-IfFailed
+
 # Test verify versions
-mkdir $VersionFilesRoot | Out-Null
-Copy-Item -Recurse "$versionFilesPath/versions_incomplete" $VersionFilesRoot
+Copy-Item -Recurse "$versionFilesPath/versions" "$RegistryRootComplete/versions"
+Copy-Item -Recurse "$versionFilesPath/ports" "$RegistryRootComplete/ports"
+& $GitExe -C "$RegistryRootComplete" add -A 
+Throw-IfFailed
+& $GitExe -C "$RegistryRootComplete" -c user.name='vcpkg-tool' -c user.email='ci@vcpkg.io' commit -m "add files"

```

Comment:
is there any reason to do this rather than `git -C ...`?

Reply:
I thought the `exe` extension is needed for windows, but it seems to be that this is not the case.